MyST codeblock parser (#27)
benji-york/manuel: src/manuel/myst-markdown.md

`Sphinx` and other `docutils` extensions provide a `code-block` directive, which allows inlined snippets of code in `MyST` documents.

Several plug-ins are included that provide new test syntax (see
{ref}`functionality`).
You can also create your own plug-ins.

For example, if you've ever wanted to include a large chunk of Python in a
doctest but were irritated by all the `>>>` and `...` prompts required, you'd
like the {mod}`manuel.myst.codeblock` module.
It lets you execute code using MyST-style `` ```python`` directives.
The markup looks like this:

    ```python
    import foo

    def my_func(bar):
        return foo.baz(bar)
    ```

To see how to get `Manuel` wired up see {ref}`getting-started`.
Be aware to use the `MyST modules` like {mod}`manuel.myst.codeblock`.

The scope of variables spans across the complete document.

```python
a = 3

# another variable
b = 2 * 3
```

The variables `a` and `b` can be used in the subsequent code block.

```python
assert b == 6
```

For a better test feedback, you can use the methods of [`unittest.TestCase`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual). 

```python
self.assertEqual(b, 6)
```

> The output of `self.assertEqual(b, 9999)` would look like

Reviewer: ```suggestion
The output of `self.assertEqual(b, 9999)` would be the following.
```

> ```shell

Reviewer: ```suggestion
```console
```

AssertionError: 6 != 9999
```

> You can even write code in invisible code blocks.

Reviewer: ```suggestion
You can even write code in [invisible code blocks](https://pythonhosted.org/manuel/#invisible-code-blocks).
```

It might also be useful to add a label above this heading in `index.txt` to make it easier to reference in markup and not require a hard link.

Reviewer: Was this suggestion overlooked?

Author: I do not see the need for this link between MyST doc and reStructuredText doc.

Reviewer: OK, I see your point. Thanks for the explanation.

Invisible code blocks do not show up in the rendered documentation.

> In fact they are comments as their lines start with `%`.

Reviewer: ```suggestion
Using MyST syntax, lines that start with `%` are comments.
```
